src/adapter/src/frontend_peek.rs

```diff
@@ -86,101 +101,195 @@ impl PeekClient {
+        let catalog = self.catalog_snapshot("try_frontend_peek").await;
 
         if let Err(_) = Coordinator::verify_portal(&*catalog, session, portal_name) {
             // TODO(peek-seq): Don't fall back to the coordinator's peek sequencing here, but retire already.
-            debug!(
-                "Bailing out from try_frontend_peek_inner, because verify_portal returned an error"
-            );
+            debug!("Bailing out from try_frontend_peek, because verify_portal returned an error");
             return Ok(None);
         }
 
-        // TODO(peek-seq): statement logging (and then enable it in various tests)
-        let (stmt, params) = {
+        // Extract things from the portal.
+        let (stmt, params, logging, lifecycle_timestamps) = {
             let portal = session
                 .get_portal_unverified(portal_name)
                 // The portal is a session-level thing, so it couldn't have concurrently disappeared
                 // since the above verification.
                 .expect("called verify_portal above");
             let params = portal.parameters.clone();
             let stmt = portal.stmt.clone();
-            (stmt, params)
-        };
-
-        let stmt = match stmt {
-            Some(stmt) => stmt,
-            None => {
-                debug!("try_frontend_peek_inner succeeded on an empty query");
-                return Ok(Some(ExecuteResponse::EmptyQuery));
-            }
+            let logging = Arc::clone(&portal.logging);
+            let lifecycle_timestamps = portal.lifecycle_timestamps.clone();
+            (stmt, params, logging, lifecycle_timestamps)
         };
 
         // Before planning, check if this is a statement type we can handle.
-        match &*stmt {
-            Statement::Select(_)
-            | Statement::ExplainAnalyzeObject(_)
-            | Statement::ExplainAnalyzeCluster(_) => {
-                // These are always fine, just continue.
-                // Note: EXPLAIN ANALYZE will `plan` to `Plan::Select`.
-            }
-            Statement::ExplainPlan(explain_stmt) => {
-                // Only handle ExplainPlan for SELECT statements.
-                // We don't want to handle e.g. EXPLAIN CREATE MATERIALIZED VIEW here, because that
-                // requires purification before planning, which the frontend peek sequencing doesn't
-                // do.
-                match &explain_stmt.explainee {
-                    mz_sql_parser::ast::Explainee::Select(..) => {
-                        // This is a SELECT, continue
-                    }
-                    _ => {
-                        debug!(
-                            "Bailing out from try_frontend_peek_inner, because EXPLAIN is not for a SELECT query"
-                        );
-                        return Ok(None);
-                    }
+        // This must happen BEFORE statement logging setup to avoid orphaned execution records.
```

Comment:
From this comment, it's not clear why doing this before statement logging setup avoids orphaned execution records.